src/snarks/groth16/mpc/chacha_rng.hpp

```diff
@@ -0,0 +1,29 @@
+#ifndef __ZETH__SNARKS_GROTH16_MPC_CHACHA_RNG_HPP__
+#define __ZETH__SNARKS_GROTH16_MPC_CHACHA_RNG_HPP__
+
+#include <cstddef>
+#include <cstdint>
+
+namespace libzeth
+{
+
+// Random number generator matching the implementation used by zcash
+// powersoftau and phase2.
```

Comment:
Can you please provide a link to the document or source file that you follow?

Reply:
Added links to the usage and implementation